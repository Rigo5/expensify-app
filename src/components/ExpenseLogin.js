import React from 'react';


class ExpenseLogin extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: ''
        }
    }

    onUsernameChange = (e) => {
        const value = e.target.value;
        
        this.setState(() => ({
            username : value
        }))
    }

    onPasswordChange = (e) => {
        const value = e.target.value;
        
        this.setState(() =>( {
            password : value
        }))
    }

    onSubmit = (e) => {
        e.preventDefault(); 
    }
    render() {
        return (<div>
            <p>Logga per accedere all'applicazione</p>
            <form onSubmit={this.onSubmit}>
                <input
                    type='text'
                    placeholder='Username'
                    value={this.state.username}
                    onChange={this.onUsernameChange}
                />
                <input
                    type='password'
                    placeholder='Password'
                    value={this.state.password}
                    onChange={this.onPasswordChange}
                />
                <button>Login</button>
            </form>
        </div>
        );
    }
}

export { ExpenseLogin as default };