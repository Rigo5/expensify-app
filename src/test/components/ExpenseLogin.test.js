import React from 'react';
import { shallow } from 'enzyme';
import ExpenseLogin from '../../components/ExpenseLogin';


test('should the ExpenseLogin render', () => {
    const wrapper = shallow(<ExpenseLogin />); 

    expect(wrapper).toMatchSnapshot(); 
})

/* test('should the ExpenseLogin render the change of the username', () => {
    const wrapper = shallow(<ExpenseLogin />);

    

    expect(wrapper.state('username')).toBe('merda'); 
}) */