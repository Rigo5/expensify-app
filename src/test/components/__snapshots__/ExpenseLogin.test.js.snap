// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should the ExpenseLogin render 1`] = `
<div>
  <p>
    Logga per accedere all'applicazione
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <button>
      Login
    </button>
  </form>
</div>
`;
